include/GafferOSL/ShadingEngine.h

```diff
@@ -79,8 +79,22 @@ class ShadingEngine : public IECore::RefCounted
 
 		IECore::CompoundDataPtr shade( const IECore::CompoundData *points, const Transforms &transforms = Transforms() ) const;
 
+		struct AttributeRead
+		{
+			AttributeRead() {}
+			AttributeRead(const std::string& scope, const std::string& name) : scope(scope), name(name) {}
+			IECore::InternedString scope;
+			IECore::InternedString name;
+		};
+
+		typedef std::vector<AttributeRead> AttributeReadContainer;
+
+		const AttributeReadContainer& getAttributesRead() const;
```

Comment:
No need for the "get" prefix since there is no matching "set". So just `attributesRead()`. Actually, can we just follow the OSL naming and call this `attributes*Needed*()` so that it's clear what the relationship to OSL is?